src/ui/developerMenu/ViewDeveloperMenu.qml

```diff
@@ -91,8 +154,9 @@ VPNFlickable {
     }
 
     VPNCheckBoxAlert {
+        id: restartRequired
         anchors.top: inspectorLink.bottom
-        visible: VPN.stagingMode != VPNSettings.stagingServer
+        visible: VPN.stagingMode != VPNSettings.stagingServer || VPN.stagingServerAddress != VPNSettings.stagingServerAddress
```

Comment:
I have a different solution. What about this:
1. `visible: false`
2.
```
Connections {
  target: VPNSettings
  function onStagingServerAddressChanged() { restartRequired.visible = true; }
  function onStagingServerChanged() { restatRequired.visible = true; }
}
```

Reply:
Your suggestion has two drawbacks: 
1. We would show the warning even if we change it back to the previous settings. 
2. If we leave the page and return it resets and we don't see the warning even if we changed something before